src/EntityFramework/Core/Query/PlanCompiler/PlanCompiler.cs

```diff
@@ -41,23 +42,68 @@ internal class PlanCompiler
     {
         #region private state
 
+        private static bool? _applyTransformationsRegardlessOfSize;
         // <summary>
-        // A boolean switch indicating whether we should apply transformation rules regardless of the size of the Iqt.
+        // A boolean value indicating whether we should apply transformation rules regardless of the size of the Iqt.
         // By default, the Enabled property of a boolean switch is set using the value specified in the configuration file.
         // Configuring the switch with a value of 0 sets the Enabled property to false; configuring the switch with a nonzero
         // value to set the Enabled property to true. If the BooleanSwitch constructor cannot find initial switch settings
         // in the configuration file, the Enabled property of the new switch is set to false by default.
         // </summary>
-        private static readonly BooleanSwitch _applyTransformationsRegardlessOfSize =
```

Comment:
I would recommend erring on the side of compatibility, so for anyone relying in the existing switch, configuration will still be applied. We can add the appSettings as a fallback.

Reply:
@divega I did restore the switch in parallel to the AppSettings